processor/src/main/java/org/mapstruct/ap/internal/model/BeanMappingMethod.java

```diff
@@ -938,6 +938,15 @@ private boolean handleDefinedMapping(MappingReference mappingRef, Type resultTyp
 
             if ( targetWriteAccessor == null ) {
                 if ( targetReadAccessor == null ) {
+                    if ( mapping.getInheritContext() != null && mapping.getInheritContext().isForwarded() &&
+                        mapping.getInheritContext().getTemplateMethod().isUpdateMethod() != method.isUpdateMethod() ) {
```

Comment:
I don't get it.. Why would you not inherit a configuration in this case? The whole idea of inheritance is that you don't have to specify mappings for update and create mappings seperately. I'm using this all the time in my own code (defining one config and then using it for both create- and update mappings. That will break now, right?

And why only forward and not reverse? That seems to me like asymetry too..

Reply:
This is an extremely special case, which was handled like this in 1.3 as well. Basically it ignores explicitly defined `@Mapping` for properties that have no write and read accessors once they are inherited. 

Take the test case. The normal mapping has write accessors for `dependantBuildRecord` and `dependantBuildRecords` (due to the builder) and you explicitly ignore both. However, the update mapping does not have an accessor for `dependantBuildRecord` and thus the `@InheritConfiguration` right now leads to an error in compilation.

Before this checks were only done when the `TargetReference` was created and those were done only on the template method. Right now they are done when handling the mapping and thus we have to explicitly handle it here.

Note: this is only done for the error handling, i.e. no target read and write accessors are available and we only ignore the "error" in this special edge case.